Nodejs/Product/Nodejs/Workspace/PackageJsonTestContainerDiscoverer.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestWindow.Extensibility;
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.Indexing;
+using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    [Export(typeof(ITestContainerDiscoverer))]
+    public sealed class PackageJsonTestContainerDiscoverer : ITestContainerDiscoverer
+    {
+        private readonly IVsFolderWorkspaceService workspaceService;
+
+        private readonly List<PackageJsonTestContainer> containers = new List<PackageJsonTestContainer>();
+        private readonly object containerLock = new object();
+
+        private IWorkspace currentWorkspace;
```

Comment:
>currentWorkspace [](start = 27, length = 16)

If `containers` is guarded by a lock, should this be as well?

Reply:
I wanted to prevent the call to `ToArray` in the `TestContainers` property to now fail, because we would be changing the collection while converting it to an array.